woocommerce/woocommerce-gateway-stripe
# feat: express checkout options migration

# Changes proposed in this Pull Request:

Fixes https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1845

Added migration for the deprecated express checkout options.

# Testing instructions

- Open the `wp_options` table in the database ( http://localhost:8083/index.php?route=/sql&db=wordpress&table=wp_options&pos=0 ) and search `option_name` for `wc_stripe_version`. Update it to an older version (e.g.: `5.3.0`)
- Disable the `_wcstripe_feature_upe_settings` feature flag, so the old settings are displayed (it's a bit easier that way)
- Visit any wp-admin page. If your payment request button type is set to any of the following (and only the following), the value should be mapped to another value:
  - `Button type: Branded` with `Branded button format: Logo only` should be updated to `Button type: Default`.
  - `Button type: Branded` with `Branded button format: Text and logo` should be updated to `Button type: Buy`.
  - `Button type: Custom` should be updated to `Button type: Buy`.
- After any visit to wp-admin the version number in the database will be updated so, after setting up the next test scenario, make sure to set the version number to something lower than `5.5.0`.
